apischeme_SSS.py

```diff
@@ -76,20 +100,10 @@ def get_bastion_ips(resource):
 
 ingress = resource.spec.managementAPIServerIngress
 
-if set(ingress.allowedCIDRBlocks) == all_ips:
+ingress_ips = set(ingress.allowedCIDRBlocks)
+if ingress_ips == all_ips:
     print("Same IPs, no-op\n%s" % all_ips)
     sys.exit(0)
 
-# Overwrite the list of IPs
-ingress.allowedCIDRBlocks = list(all_ips)
-print("Applying IPs: %s" % ingress.allowedCIDRBlocks)
-
-# Tell cloud-ingress-operator it's okay to apply the CIDRs now.
-if not ingress.enabled:
-    print("Enabling ingress")
-    ingress.enabled = True  # As opposed to the string "true".
-
-sss_resources = dyn_client.resources.get(
-    api_version="hive.openshift.io/v1", kind="SelectorSyncSet"
-)
-dyn_client.apply(sss_resources, body=sss.to_dict())
+missing_ips = [ip for ip in all_ips if ip not in ingress_ips]
+add_missing_ips(missing_ips)
```

Comment:
Since they're already sets, this could be

```suggestion
missing_ips = all_ips - ingress_ips
add_missing_ips(list(missing_ips))
```

However, I think there's a bug here: `all_ips - ingress_ips` can still be empty when `ingress_ips == all_ips` is `False` -- e.g. if `ingress_ips` is a superset of `all_ips`. It looks like we're currently not trying to propose MRs to _remove_ IPs (what's the plan for that?). So we should really be collapsing these conditions:

```python
ingress_ips = set(ingress.allowedCIDRBlocks)
missing_ips = all_ips - ingress_ips
if not missing_ips:
    print("No IPs to add, no-op")
    sys.exit(0)

add_missing_ips(list(missing_ips))
```

Reply:
done